src/tools/wast2json.cc

```diff
@@ -115,12 +116,23 @@ int ProgramMain(int argc, char** argv) {
     }
 
     if (Succeeded(result)) {
-      WriteBinarySpecOptions write_binary_spec_options;
-      write_binary_spec_options.log_stream = s_log_stream.get();
-      write_binary_spec_options.json_filename = s_outfile;
-      write_binary_spec_options.write_binary_options = s_write_binary_options;
-      result = WriteBinarySpecScript(script.get(), s_infile,
-                                     &write_binary_spec_options);
+      std::vector<FilenameMemoryStreamPair> module_streams;
+      MemoryStream json_stream;
+
+      std::string module_filename_noext =
+          StripExtension(s_outfile ? s_outfile : s_infile).to_string();
+      result = WriteBinarySpecScript(
+          &json_stream, script.get(), s_infile, module_filename_noext,
+          &s_write_binary_options, &module_streams, s_log_stream.get());
+
+      if (s_outfile) {
+        json_stream.WriteToFile(s_outfile);
+      }
+
+      std::for_each(module_streams.begin(), module_streams.end(),
```

Comment:
Is this preferable to using a normal for loop?    Seems harder to visually parse and more verbose.